base/serialization.cpp

```diff
@@ -26,6 +26,10 @@ BinStream::BinStream(size_t sz) : front_(0) { buffer_.resize(sz); }
 
 BinStream::BinStream(const char* src, size_t sz) : front_(0) { push_back_bytes(src, sz); }
 
+BinStream::BinStream(const std::vector<char>& v) : front_(0) { buffer_ = v; }
+
+BinStream::BinStream(std::vector<char>&& v) : front_(0) { buffer_ = std::move(v); }
```

Comment:
`buffer_ = v` and `buffer_ = std::move(v)` can be initialized in initialization list.